noxfile-template.py

```diff
@@ -17,6 +17,7 @@
 import os
 from pathlib import Path
 import sys
+from typing import Callable, Dict, List, Optional
```

Comment:
noxfile-template.py shouldn't be committed - the noxfile is autogenerated from the root `noxfile-template.py`

Reply:
UPDATE I totally misread what this was - this is in fact @glasnt being awesome and adding type hinting HERE so that we can actually enforce type hints in samples, which is something I missed doing in the first place.

Reply:
And, by doing it here, we can test against this brand new sample!! yay!